From 82daba7ad2fab059b8091040ae60947bcaee2a7b Mon Sep 17 00:00:00 2001
From: Kihwal Lee <kihwal@apache.org>
Date: Tue, 17 Jan 2017 15:13:56 -0600
Subject: [PATCH 2249/2848] HADOOP-13976. Path globbing does not match
 newlines. Contributed by Eric Badger.

(cherry picked from commit 706d630eb9db9658083d57d1d99b6a0f11cc5657)
(cherry picked from commit 7d949cc43250e1c0eae7ba8864511be0340c3e97)

Change-Id: I2d8b17ab26b68a34335e5a4586e4f7869cce20f5
---
 .../java/org/apache/hadoop/fs/GlobPattern.java     |    2 +-
 .../java/org/apache/hadoop/fs/TestGlobPattern.java |    8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/GlobPattern.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/GlobPattern.java
index 4be5b1c..c214609 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/GlobPattern.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/GlobPattern.java
@@ -153,7 +153,7 @@ public void set(String glob) {
     if (curlyOpen > 0) {
       error("Unclosed group", glob, len);
     }
-    compiled = Pattern.compile(regex.toString());
+    compiled = Pattern.compile(regex.toString(), Pattern.DOTALL);
   }
 
   /**
diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestGlobPattern.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestGlobPattern.java
index 0fffc47..128ac23 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestGlobPattern.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestGlobPattern.java
@@ -50,11 +50,11 @@ private void shouldThrow(String... globs) {
   }
 
   @Test public void testValidPatterns() {
-    assertMatch(true, "*", "^$", "foo", "bar");
+    assertMatch(true, "*", "^$", "foo", "bar", "\n");
     assertMatch(true, "?", "?", "^", "[", "]", "$");
-    assertMatch(true, "foo*", "foo", "food", "fool");
-    assertMatch(true, "f*d", "fud", "food");
-    assertMatch(true, "*d", "good", "bad");
+    assertMatch(true, "foo*", "foo", "food", "fool", "foo\n", "foo\nbar");
+    assertMatch(true, "f*d", "fud", "food", "foo\nd");
+    assertMatch(true, "*d", "good", "bad", "\nd");
     assertMatch(true, "\\*\\?\\[\\{\\\\", "*?[{\\");
     assertMatch(true, "[]^-]", "]", "-", "^");
     assertMatch(true, "]", "]");
-- 
1.7.9.5

